gramps/plugins/tool/reorderids.py

```diff
@@ -585,8 +591,12 @@ def _reorder(self, prim_obj):
         change = self.obj_values[prim_obj].get_change()
         index_max = int("9" * self.obj_values[prim_obj].width_fmt)
         do_same = False
+        # Process in handle order, which is in order handles were created.
```

Comment:
This is a side-effect of the way we currently generate handles.  It should not be assumed that this will be true on all database backends.  We may even choose to use a UUID as a handle in future versions.

Reply:
I still think this is a reasonable idea to allow a full reorder/renumber to maintain pretty much the same order as items were originally assigned.  Even if we switch to some other handle method, I believe that many UUIDs include a time element in their creation which would make sorting possible.

At the very least, sorting here makes it such that multiple invocations of the Reorder tool don't get vastly different results.

Reply:
Sorting the handle to make the output idempotent is acceptable.

If we choose to use a UUID for the handle we would use `uuid4()` from the python [uuid](https://docs.python.org/3.5/library/uuid.html) module.  At the same time we could extract a creation timestamp from the old handle format if necessary.